src/handlers/GraphQLHandler.ts

```diff
@@ -125,6 +125,11 @@ export class GraphQLHandler<
     }
   }
 
+  public test(request: MockedRequest): boolean {
+    const hasMatchingUrl = matchRequestUrl(request.url, this.endpoint)
+    return hasMatchingUrl.matches && super.test(request)
```

Comment:
The endpoint URL match is already implemented in the `predicate` method. The parent class calls `this.predicate` as a part of the `test` method:

https://github.com/mswjs/msw/blob/1da34e4fef13e278043d65ad656875186ac84e90/src/handlers/GraphQLHandler.ts#L128-L157

https://github.com/mswjs/msw/blob/1da34e4fef13e278043d65ad656875186ac84e90/src/handlers/RequestHandler.ts#L158-L160

Internally, parsing always precedes the predicate, because the request handler's predicate may depend on some request data available only after the request has been parsed (even if it's an irrelevant request—that's how the relevance is determined).